Lib/test/test_typing.py

```diff
@@ -4408,6 +4408,12 @@ def test_get_type_hints_typeddict(self):
             'a': Annotated[Required[int], "a", "b", "c"]
         })
 
+    def test_get_type_hints_collections_abc_callable(self):
+        # https://github.com/python/cpython/issues/91621
+        def f(x: collections.abc.Callable[[int], int]): ...
+
+        self.assertEqual(get_type_hints(f), {'x': collections.abc.Callable[[int], int]})
```

Comment:
Can you add tests for some edge cases like `Callable[..., int]` and `Callable[P, int]`?